backend/src/controllers/BaseController.js

```diff
@@ -0,0 +1,87 @@
+import ApiError from '../utils/ApiError.js';
+import asyncHandler from '../utils/asyncHandler.js';
+
+class BaseController {
+  constructor(service) {
+    if (!service) {
+      throw new ApiError(500, 'Service instance is required for BaseController');
+    }
+    this.service = service;
+  }
+
+  getAll = asyncHandler(async (req, res) => {
+    const result = await this.service.getAll();
+    res.status(200).json({
+      success: true,
+      data: result,
+    });
+  });
```

Comment:
Why you're using this asyncHandler when you can just simply use it as async method like below.

```suggestion
  async getAll(req, res) {
    const result = await this.service.getAll();
    res.status(200).json({
      success: true,
      data: result,
    });
  };
```

Reply:
Bhaia I have created a promise based `asyncHandler` wrapper for the controllers to use. The main reason to do this is to avoid using `try..catch` everywhere.